banking/ebics/doctype/ebics_user/ebics_user.py

```diff
@@ -0,0 +1,124 @@
+# Copyright (c) 2024, ALYF GmbH and contributors
+# For license information, please see license.txt
+
+import frappe
+from frappe import _
+from frappe.model.document import Document
+from frappe.utils import get_link_to_form
+
+from banking.ebics.utils import get_ebics_manager, sync_ebics_transactions
+from banking.klarna_kosma_integration.admin import Admin
+
+
+class EBICSUser(Document):
+	def validate(self):
+		if self.country:
+			self.validate_country_code()
+
+		if self.bank:
+			self.validate_bank()
+
+	def before_insert(self):
+		self.register_user()
+
+	def on_update(self):
+		self.register_user()
+
+	def on_trash(self):
+		self.remove_user()
+
+	def register_user(self):
+		"""Indempotent method to register the user with the admin backend."""
+		host_id = frappe.db.get_value("Bank", self.bank, "ebics_host_id")
+		Admin().request.register_ebics_user(host_id, self.partner_id, self.user_id)
+
+	def remove_user(self):
+		"""Indempotent method to remove the user from the admin backend."""
+		host_id = frappe.db.get_value("Bank", self.bank, "ebics_host_id")
+		Admin().request.remove_ebics_user(host_id, self.partner_id, self.user_id)
+
+	def validate_country_code(self):
+		country_code = frappe.db.get_value("Country", self.country, "code")
+		if not country_code or len(country_code) != 2:
+			frappe.throw(
+				_("Please add a two-letter country code to country {0}").format(
+					get_link_to_form("Country", self.country)
+				)
+			)
```

Comment:
we can make sure the country code is mandatory in frappe ? Are there cases where this asserts?

Reply:
https://github.com/frappe/frappe/pull/28307